lib/caasp.pm

```diff
@@ -158,10 +158,27 @@ sub check_reboot_changes {
 sub trup_install {
     # rebootmgr has to be turned off as prerequisity for this to work
     script_run "rebootmgrctl set-strategy off";
-    my $package = shift;
-    trup_call("pkg install $package");
-    process_reboot 1;
-    assert_script_run("rpm -qi $package");
+    my ($input, $unnecessary, $necessary, $pkg_is_present) = shift;
```

Comment:
`shift` will pass only first argument, so if you just want to declare more variables, I would suggest to split it. And potentially move part with passed parameters to the first line (that's just matter of style).

Reply:
Do you mean:

```perl
my $input = shift;
my $unnecessary = shift;
my $necessary = shift;
my $pkg_is_present = shift;
```

can I do the same without using `shift` at all?

e.g.
my ($input, $unnecessary, $necessary, $pkg_is_present);

Reply:
```perl
my ($input, $unnecessary, $necessary, $pkg_is_present) = @_;
```

Reply:
Sorry, I propaply misunderstood your intention. Construct `my ($a, $b, $c) = @_;` assigns passed parameters to each variable in that order.
You pass only 1 parameter, $input. You probably wanted:
```perl
# Take 1 parameter passed to function
my $input = shift;

# Define local variables
my ($unnecessary, $necessary, $pkg_is_present);
my ($pkg_required, $not_needed);
```